src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java

```diff
@@ -257,6 +315,55 @@ else if (otherValue == null)
             }
             return operator.isSatisfiedBy(type, otherValue, value);
         }
+
+        void checkForUnsetValues(List<ByteBuffer> values)
```

Comment:
With this in place, do we still need the per-value checks in `compareWithOperator()`? I think the only thing we're not currently double-validating is the usage in `TxnCondition#applies()`.

Reply:
It's probably redundant, but that is such a fast thing to check at runtime I think it's fine to do it in both?

I added this check and moved it earlier because I was unsure if we had those unset values in practice and in which types of bounds. It's a little easier to find out the unset value is unset when the bound is constructed rather than later when it is applied.

Don't feel strongly either way.

Reply:
> It's a little easier to find out the unset value is unset when the bound is constructed rather than later when it is applied.

Would it make sense to have the `value == UNSET_BYTE_BUFFER` check in the `TxnCondition.Value` constructor? We'd find out earlier I guess.

In terms of whether we'd want to remove the duplicate checks in `compareWithOperator()`, I'll leave it up to you. If I'm able to make it private again when I finish the CQL integration, perhaps I'd remove them, but I'm fine either way at this point.

Reply:
For `CQL3CasRequest` this would be later not earlier? The bounds are created when the request is first created from the parser/planner bits.